drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -424,12 +430,83 @@ int lis2dh_init(const struct device *dev)
 		    cfg->bus_name, 1 << (LIS2DH_FS_IDX + 1),
 		    LIS2DH_ODR_IDX, (uint8_t)LIS2DH_LP_EN_BIT, lis2dh->scale);
 
+#ifdef CONFIG_PM_DEVICE
+	lis2dh->power_state = PM_DEVICE_STATE_ACTIVE;
+	lis2dh->reg_ctrl1_active_val = LIS2DH_ACCEL_EN_BITS | LIS2DH_LP_EN_BIT |
+				       LIS2DH_ODR_BITS;
+#endif
+
 	/* enable accel measurements and set power mode and data rate */
 	return lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL1,
 					LIS2DH_ACCEL_EN_BITS | LIS2DH_LP_EN_BIT |
 					LIS2DH_ODR_BITS);
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int lis2dh_set_power_state(const struct device *dev,
+				  enum pm_device_state new_state)
+{
+	struct lis2dh_data *lis2dh = dev->data;
+	int status = 0;
+
+	if (new_state == PM_DEVICE_STATE_ACTIVE) {
+		status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL1,
+						  lis2dh->reg_ctrl1_active_val);
+		if (status) {
+			LOG_ERR("setting active state failed");
+		}
+		lis2dh->power_state = PM_DEVICE_STATE_ACTIVE;
+		LOG_INF("state changed to active");
+	} else {
+		__ASSERT_NO_MSG(new_state == PM_DEVICE_STATE_LOW_POWER ||
+				new_state == PM_DEVICE_STATE_SUSPEND ||
+				new_state == PM_DEVICE_STATE_OFF);
+		status = lis2dh->hw_tf->read_reg(dev, LIS2DH_REG_CTRL1,
+						 &lis2dh->reg_ctrl1_active_val);
+		if (status) {
+			LOG_ERR("failed to read reg_crtl1");
+			return -EIO;
+		}
+		status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL1,
+						  LIS2DH_POWER_DOWN);
+		if (status) {
+			LOG_ERR("failed to write power down command");
+			return -EIO;
+		}
+		lis2dh->power_state = new_state;
+		LOG_INF("State changed to inactive");
+	}
+
+	return status;
+}
+
+static int lis2dh_pm_control(const struct device *dev, uint32_t ctrl_command,
+				enum pm_device_state *state)
+{
+	struct lis2dh_data *lis2dh = dev->data;
+	enum pm_device_state current_state = lis2dh->power_state;
+	int status = 0;
+	enum pm_device_state new_state;
+
+	switch (ctrl_command) {
+	case PM_DEVICE_STATE_SET:
+		new_state = *state;
+		if (new_state != current_state) {
+			status = lis2dh_set_power_state(dev, new_state);
+		}
+		break;
+	case PM_DEVICE_STATE_GET:
+		*state = current_state;
+		break;
+	default:
+		LOG_ERR("Got unknown power management control command");
+		status = -EINVAL;
+	}
+
+	return status;
+}
+#endif /* CONFIG_PM_DEVICE */
```

Comment:
this seems to be outdated, how have you tested this with `main`?

Reply:
Bringing this forward from a slightly older project, hadn't realized PM functions had changed so much. Updating.